apps/web/components/connections/SourceCard/SourceCard.tsx

```diff
@@ -0,0 +1,279 @@
+              onUpdateSuccess={() => setIsRenaming(false)}
+            />
+          ) : (
+            <span className="truncate text-sm font-medium uppercase">
+              {displayName}
+            </span>
+          )}
+
+          <ActionMenu
+            isOpen={isActionMenuOpen}
+            onOpenChange={setActionMenuOpen}>
+            <ActionMenuItem
+              icon={EditIcon}
+              label="Rename"
+              onClick={() => setIsRenaming(true)}
+            />
+            <ActionMenuItem
+              icon={SyncIcon}
+              label="Sync"
+              // TODO: show sync in progress and result (success/failure)
+              onClick={() =>
+                dispatch.mutate({
+                  name: 'resource/sync-requested',
+                  data: {resourceId: id},
+                })
+              }
+            />
+            <ActionMenuItem
+              icon={DeleteIcon}
+              label="Delete"
+              onClick={() => {
+                // need to close dropdown menu first
+                setActionMenuOpen(false)
+                // ...
+                setTimeout(() => setDeleteDialogOpen(true), 0)
+              }}
+            />
+          </ActionMenu>
+
+          {/* Needs to keep Dialog root outside ActionMenu otherwise it won't open */}
+          <Dialog.Root
+            open={isDeleteDialogOpen}
+            onOpenChange={setDeleteDialogOpen}>
+            <DeleteSourceDialog
+              resourceId=""
+              name="Wells Fargo"
+              institution={{
+                name: 'Wells Fargo',
+                logoUrl: '/wells-fargo-logo.svg',
+              }}
+              onCancel={() => setDeleteDialogOpen(false)}
+            />
+          </Dialog.Root>
+        </div>
+        <div className="text-right">
+          {status ? <ConnectionStatus status={status} /> : null}
+          <p className="text-xs font-medium text-venice-gray">
+            {lastSyncCompletedAt
+              ? `Synced ${formatDistanceToNowStrict(
+                  new Date(lastSyncCompletedAt),
+                  {addSuffix: true},
+                )}`
+              : 'No sync information'}
+          </p>
+        </div>
+      </div>
+    </ResourceCard>
+  )
+}
+
+interface EditingDisplayNameProps {
+  displayName: string
+  onCancel: () => void
+  onUpdateSuccess: () => void
+  resourceId: string
+}
+
+function EditingDisplayName(props: EditingDisplayNameProps) {
+  const {onCancel, onUpdateSuccess, resourceId} = props
+  const [displayName, setDisplayName] = useState(props.displayName)
+  const updateResource = mutations.useUpdateResource()
+
+  useEffect(() => {
+    async function handleKeyUp(event: KeyboardEvent) {
+      switch (event.key) {
+        case 'Escape':
+          onCancel()
+          break
+        case 'Enter':
+          await updateResource.mutateAsync({
+            // @ts-expect-error fix query or mutation type
```

Comment:
@tonyxiao type mismatch between the mutation and query (`string` vs `reso${...}`